src/librustc/middle/const_eval.rs

```diff
@@ -90,7 +91,8 @@ fn lookup_variant_by_id<'a>(tcx: &'a ty::ctxt,
 
 pub fn lookup_const_by_id<'a, 'tcx: 'a>(tcx: &'a ty::ctxt<'tcx>,
                                         def_id: DefId,
-                                        maybe_ref_id: Option<ast::NodeId>)
+                                        maybe_ref_id: Option<ast::NodeId>,
+                                        param_substs: Option<&'tcx subst::Substs<'tcx>>)
```

Comment:
This merits a comment, I think, as to the purpose of each of these parameters. Something like:

- `def_id` is the id of the constant
- `maybe_ref_id` is the id of the expression referencing the constant, if available
- `param_substs` is the monomorphization substitution for the expression, if available

The latter two parameters are used to extract the substitutions for the constant, in the case of associated constants.